api/utils/sshutils/ssh.go

```diff
@@ -174,12 +175,17 @@ func AsAgentKeys(sshCert *ssh.Certificate, privKey []byte) ([]agent.AddedKey, er
 func HostKeyCallback(caCerts [][]byte) (ssh.HostKeyCallback, error) {
 	var trustedKeys []ssh.PublicKey
 	for _, caCert := range caCerts {
-		_, _, publicKey, _, _, err := ssh.ParseKnownHosts(caCert)
-		if err != nil {
-			return nil, trace.BadParameter("failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+		var err error
+		for err == nil {
+			var publicKey ssh.PublicKey
+			_, _, publicKey, _, caCert, err = ssh.ParseKnownHosts(caCert)
+			if err != nil && err != io.EOF {
+				return nil, trace.Wrap(err, "failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+			}
+			trustedKeys = append(trustedKeys, publicKey)
```

Comment:
On a last iteration when `err == io.EOF` this would append a `nil` to `trustedKeys` which could be troublesome.

Reply:
:facepalm: